includes/class-plugin-autoupdate-filter-helpers.php

```diff
@@ -0,0 +1,138 @@
+<?php
+/**
+ * Plugin Autoupdate Filter Helpers class
+ *
+ * @package Plugin_Autoupdate_Filter
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly
+}
+
+class Plugin_Autoupdate_Filter_Helpers {
+
+	/**
+	 * Determines whether a plugin should be updated based on release version and delay rules.
+	 *
+	 * @param string $plugin_slug Slug of the plugin.
+	 * @param string $update_version The version that the plugin would be updated to.
+	 *
+	 * @return bool True if the plugin should be updated, false otherwise.
+	 */
+	public function has_delay_passed( string $plugin_slug, string $update_version ): bool {
+		// delay most plugins 2 days. delay woocommerce 7 days.
+		$delay_days        = 'woocommerce' === $plugin_slug ? 7 : 2;
+		$installed_version = $this->get_installed_plugin_version( $plugin_slug );
+
+		if ( empty( $installed_version ) || $update_version === $installed_version ) {
+			return false;
+		}
+
+		$installed_version_parts = explode( '.', $installed_version );
+		$update_version_parts    = explode( '.', $update_version );
+
+		if ( 3 !== count( $installed_version_parts ) || 3 !== count( $update_version_parts ) ) {
```

Comment:
Interesting! Doesn't this mean we will never install versions that only have two digits on their numbering?
For instance - Yoast: https://developer.yoast.com/changelog/yoast-seo/ or

Reply:
I didn't think about plugins that use different numbering styles. Looking at Yoast, I see they use both, e.g. they released `21.8` and then `21.8.1` right after that. 

I'll need to account for this.

Reply:
I think this actually works as intended if I just remove that line that checks for 3 numbers in the version. My following code still assumes that the first number is the major version, and the second number is the minor version, and I believe that still holds true for plugins which only use 2 numbers in their version.

Reply:
Side note: I also saw a plugin today that uses this wild version: `Sensei Pro (WC Paid Courses) version 4.24.0.1.24.0`